### ISSUE TITLE
Pre-populate checkout form customer email via URL query string

### ISSUE THREAD

**Issue by davidsherlock** [easydigitaldownloads/easy-digital-downloads#7104]
### Explain your enhancement (please be detailed)

A customer has suggested that we introduce a way to pre-populate the checkout email field to make the checkout process more seamless.

The following example was given:

> 1. People download our plugin from WordPress.org as well as our site.
> 2. For those people that download the free version (from either place), we want to market to them.
> 3. We are going to have some sections in our plugin that require an add on to be purchased in order to add that functionality.
> 4. So, when a user of the plugin goes to our "Import and Export" page for example, they would be told that they need to purchase that addon from our site.
> 5. We will use the documentation here to bring them to the cart / checkout page with the download URL and automatically added to the cart:
> 
> https://docs.easydigitaldownloads.com/article/268-creating-custom-add-to-cart-links
> 
> So, the URL we will use will look like this:
> 
> https://www.wpinventory.com/checkout?edd_action=add_to_cart&download_id=655
> 
> 6. And, here is where we want to take it one step further. We want to attach the 'admin_email' from the wordpress options table in the URL:
> 
> https://www.wpinventory.com/checkout?edd_action=add_to_cart&download_id=655&email=adminemail@emailprovider.com
> 
> 7. If we have that $_GET variable, then we will populate the input value with it. If not, default to nothing.
> 
> I have not seen any documentation on how to change this / alter the personal information data. The only thing in the general area I was able to dig up was altering the checkout_cart.php via template override. But, that is not the full page; just the top table of actions.

### Justification or use case

Make the checkout process more efficient when sending new customers from your plugin or similar.

As a side note, I feel this should _potentially_ be a code snippet in the library.

**Comment by pippinsplugins** [comment]
I like this a lot. It's a great way to help make checkout more seamless when customers land on checkout from somewhere else where the email is known.

**Comment by cklosowski** [comment]
Yeah this could make it easier, sholdn't take too much work. A few things we'll have to keep in mind:
- Proper sanitization to keep things safe
- If the user comes to the page logged in, I think we should just rely on the customer account that already exists, not what's in the query string, this will avoid any weird discrepancy in the customer account.
- Odds are this will end up in a template change, so we'll have to be clear about that in the changelog that it's a change to the template files.

**Comment by wpinventory** [comment]
Hey guys!  Thank you VERY much for putting this up and taking it on.  I appreciate you taking this suggestion seriously :) . Do you have an ETA / release version this might go into effect?

Thank you very very much!

**Comment by cklosowski** [comment]
@wpinventory it's currently slated for version 3.1. we are nearing the end of the 3.0 dev cycle right now

**Comment by cklosowski** [comment]
Some further thinking on this, I think we need to account for a couple things:
1. It might be helpful to also accept first name and last name.
2. Regardless of the fields we auto-fill based on the query string, we need to be sure to trigger the `on change` Javascript event, so that any customizations or additional JS added to checkout are run.

**Comment by cklosowski** [comment]
Some more things to consider as I look at this:

If the user is logged in, do we populate the field with email address? Or do we rely on the logged in user email address as we currently do?

**Comment by cklosowski** [comment]
While I like this idea, I think we need to wait. With some recent issues with people changing email addresses at checkout and it producing new customers, as well as some changes to the checkout coming in the form of blocks, I want to revisit this once we have some time to live with the new models we're releasing with 3.0 and the new checkout.
